tools/structure/src/main.py

```diff
@@ -156,6 +155,18 @@ def run(
                     reindex = False
             except Exception as e:
                 self.stream_error_and_exit(f"Error fetching data and indexing: {e}")
+
+            for output in outputs:
+                try:
+                    table_settings = output[SettingsKeys.TABLE_SETTINGS]
+                    extracted_input_file = tool_data_dir / SettingsKeys.EXTRACT
+                    table_settings[SettingsKeys.INPUT_FILE] = extracted_input_file
+                    output.update({SettingsKeys.TABLE_SETTINGS: table_settings})
+
+                except KeyError:
+                    # To check if the prompt has table enforce type selected.
+                    pass
```

Comment:
@Deepak-Kesavan @harini-venkataraman can this be pluggable too?

Reply:
@chandrasekharan-zipstack We can do it. But I feel that it won't be an issue even if it is left like this. We won't be processing it if table extraction plugin is not there in prompt-service

Reply:
@chandrasekharan-zipstack We dont expose much logic here, and didn't want to make it pluggable since it is in tools. It will not be executed if the table is not chosen.

Reply:
My suggestion would be to add a TODO comment and create a ticket to mark it as a tech debt. We aren't tracking any such code that go in under this pretext. I feel that this is how cloud related logic creeps into the OSS code.
Even if something doesn't impact the OSS logic directly, it does make us question what a code snippet does and we'll not have a valid answer for that.
Feel free to resolve this conversation

Reply:
@chandrasekharan-zipstack Thanks for the suggestion. Will track this under TODO for pluggability in tools.